txscript/sigcache.go

```diff
@@ -141,3 +146,60 @@ func shortTxHash(msg *wire.MsgTx, key [shortTxHashKeySize]byte) uint64 {
 	txHash := msg.TxHash()
 	return siphash.Hash(k0, k1, txHash[:])
 }
+
+// EvictEntries removes all entries from the SigCache that correspond to the
+// transactions in the given block.  The block that is passed must be
```

Comment:
Don't change this on my account, but "must" here feels overly defensive. Nothing critically bad (by that I mean errors, panics, undefined behavior, etc) happens if you pass a block that is _not_ `ProactiveEvictionDepth`: you just get a less useful cache (up to potentially _useless_), but that's a problem for the SigCache's client code to decide.

Reply:
Don't have a strong opinion here but updated the wording to `should` instead of `must`. Thanks for the review!